pkg/apis/jenkins.io/v1/types_users.go

```diff
@@ -45,6 +45,7 @@ type UserDetails struct {
 	AvatarURL         string             `json:"avatarUrl,omitempty"  protobuf:"bytes,6,opt,name=avatarUrl"`
 	ServiceAccount    string             `json:"serviceAccount,omitempty"  protobuf:"bytes,7,opt,name=serviceAccount"`
 	Accounts          []AccountReference `json:"accountReference,omitempty"  protobuf:"bytes,8,opt,name=accountReference"`
+	CreateServiceAc   bool               `json:"createServiceAc,omitempty"  protobuf:"bytes,9,opt,name=createServiceAc"`
```

Comment:
I would use full words for this for clarity since is an API definition, e.g `CreateServiceAccount`.

Reply:
First comment - done.
Second comment - I'm not sure I follow: the `ServiceAccount` property will _always_ be empty when the User is first created, until the controller creates the `ServiceAccount` then updates the `User` to set this reference. i.e. the controller would _always_ create the ServiceAccount, rather than selected during user creation.

Reply:
Not sure then, If I understand the purpose of `CreateServiceAccount` flag. What's going to happen if it is `false`? Will the controller not create the user?

Reply:
Yes, that's right: if `CreateServiceAccount==false`, then the controller won't do anything. Only if `CreateServiceAccount==true` will it create the `ServiceAccount` and set the ref. into `User.ServiceAccount`.

Reply:
What is going to happen with the users that do not have a service account? Is the main purpose of this flag to delay the service account creation until an admin comes and set it to `true`? TBH I don't have an overview of the entire use case. 

My only concern is that we will have a user resource with status `created` but with inconsistent metadata. Some users will have service accounts while others not. Probably, we should define also different statuses for user resource e.g. init, ready etc. 

Normally, a controller should watch when a user is created, create the service account and update the status of the users as e.g. `provisioned/ready`.

Reply:
users who are known about by the k8s platform do not need to have service accounts.
for example if using gke and you have a basic role (enough to be able to use gcloud login) to access the cluster then jenkins x needs the User in order to be able to work out who to apply the roles to.
in the case of a Google Cloud User they do not need a service account.  For non cloud users (ie those without access to the k8s cluster) they need the service account.

//cc @swashbuck1r

Reply:
I think we are just getting confused about the name of this.  to me is a boolean flag to indicate the type of user - perhaps `nativeUser` would be more correct - and that would be true for users that auth with the cluster and false for ones that need some other form of auth (like the service account)

Reply:
Agreed, that seems more consistent with the way these APIs work.

Reply:
@ccojocar - just to address your questions: metadata shouldn't be inconsistent; we will (well, could) have a mix of users with & without a ServiceAccount, but only because they are of different types, some being 'local' K8S users (no SA) and some are external users (with SA). The logic and state should be consistent, whereby the controller will always ensure SA exists for those users that need it. There's no expectation that an admin or anyone else would ever change the state of this field: it is set at user creation and represents the type of user (SA required or not), and _that never changes_.